common/djangoapps/student/tests/test_models.py

```diff
@@ -244,6 +256,150 @@ def test_enrollments_not_deleted(self):
+        celebration, should_celebrate = CourseEnrollmentCelebration.should_celebrate_streak(enrollment)
+        today = datetime.datetime.now(UTC).date()
+        self.assertEqual(celebration.first_day_of_streak, today)
+        self.assertEqual(celebration.last_day_of_streak, today)
+        self.assertEqual(celebration.last_day_of_streak, today)
+        self.assertEqual(should_celebrate, False)
+
+    # lint-amnesty, pylint: disable=line-too-long
+    def test_celebrate_only_once_in_continuous_streak(self):
+        """
+        Sample run for a 3 day streak and 1 day break. See last column for explanation.
+        +---------+---------------------+--------------------+-------------------------+------------------+-----------------------------------------------+
+        | today   | first_day_of_streak | last_day_of_streak | last day of celebration | should_celebrate | Note                                          |
+        +---------+---------------------+--------------------+-------------------------+------------------+-----------------------------------------------+
+        | 2/4/21  | 2/4/21              | 2/4/21             | None                    | FALSE            | Day 1 of Streak                               |
+        | 2/5/21  | 2/4/21              | 2/5/21             | None                    | FALSE            | Day 2 of Streak                               |
+        | 2/6/21  | 2/4/21              | 2/6/21             | 2/6/21                  | TRUE             | Completed 3 Day Streak so we should celebrate |
+        | 2/7/21  | 2/4/21              | 2/7/21             | 2/6/21                  | FALSE            | Day 4 of Streak                               |
+        | 2/8/21  | 2/4/21              | 2/8/21             | 2/6/21                  | FALSE            | Day 5 of Streak                               |
+        | 2/9/21  | 2/4/21              | 2/9/21             | 2/6/21                  | FALSE            | Day 6 of Streak                               |
+        +---------+---------------------+--------------------+-------------------------+------------------+-----------------------------------------------+
+        """
+        enrollment = CourseEnrollmentFactory(course_id=self.course.id)  # pylint: disable=no-member
+        today = datetime.datetime.now(UTC)
+        for i in range(1, (STREAK_LENGTH_TO_CELEBRATE * 2) + 1):
+            with mock.patch.object(CourseEnrollmentCelebration, '_get_today') as get_today_mock:
+                get_today_mock.return_value = today + datetime.timedelta(days=i)
+                _, should_celebrate = CourseEnrollmentCelebration.should_celebrate_streak(enrollment)
+                self.assertEqual(should_celebrate, i == STREAK_LENGTH_TO_CELEBRATE)
+
+    # lint-amnesty, pylint: disable=line-too-long
+    def test_celebrate_only_once_with_multiple_calls_on_the_same_day(self):
```

Comment:
the sample runs should be readable in full width (if you use the github unified diff mode under the settings gear or click the three dots in top right and click view file)

Reply:
Tables are nice, thanks.